planning/behavior_path_planner/src/scene_module/dynamic_avoidance/dynamic_avoidance_module.cpp

```diff
@@ -0,0 +1,392 @@
+  const std::string & name, rclcpp::Node & node,
+  std::shared_ptr<DynamicAvoidanceParameters> parameters)
+: SceneModuleInterface{name, node, createRTCInterfaceMap(node, name, {""})},
+  parameters_{std::move(parameters)}
+#else
+DynamicAvoidanceModule::DynamicAvoidanceModule(
+  const std::string & name, rclcpp::Node & node,
+  std::shared_ptr<DynamicAvoidanceParameters> parameters,
+  const std::unordered_map<std::string, std::shared_ptr<RTCInterface> > & rtc_interface_ptr_map)
+: SceneModuleInterface{name, node, rtc_interface_ptr_map}, parameters_{std::move(parameters)}
+#endif
+{
+}
+
+bool DynamicAvoidanceModule::isExecutionRequested() const
+{
+  RCLCPP_DEBUG(getLogger(), "DYNAMIC AVOIDANCE isExecutionRequested.");
+
+  // check if the ego is driving forward
+  const auto is_driving_forward = [&]() {
+    if (!getPreviousModuleOutput().path || getPreviousModuleOutput().path->points.size() < 2) {
+      return false;
+    }
+    const auto is_driving_forward =
+      motion_utils::isDrivingForward(getPreviousModuleOutput().path->points);
+    if (!is_driving_forward) {
+      return false;
+    }
+    return *is_driving_forward;
+  }();
+
+  // check if the planner is already running
+  if (current_state_ == ModuleStatus::RUNNING) {
+    return true;
+  }
+
+  // check if there is target objects to avoid
+  return !target_objects_.empty();
+}
+
+bool DynamicAvoidanceModule::isExecutionReady() const
+{
+  RCLCPP_DEBUG(getLogger(), "DYNAMIC AVOIDANCE isExecutionReady.");
+  return true;
+}
+
+void DynamicAvoidanceModule::updateData()
+{
+  target_objects_ = calcTargetObjects();
+}
+
+ModuleStatus DynamicAvoidanceModule::updateState()
+{
+  const bool has_avoidance_target = !target_objects_.empty();
+
+  if (!has_avoidance_target) {
+    current_state_ = ModuleStatus::SUCCESS;
+  } else {
+    current_state_ = ModuleStatus::RUNNING;
+  }
+
+  return current_state_;
+}
+
+BehaviorModuleOutput DynamicAvoidanceModule::plan()
+{
+  const auto & p = planner_data_->parameters;
+
+  // 1. generate reference path
+  auto reference_path = utils::generateCenterLinePath(planner_data_);
+
+  // 2. generate drivable lanes
+  const auto current_lanes = utils::getCurrentLanes(planner_data_);
+  const auto drivable_lanes = utils::generateDrivableLanes(current_lanes);
+  const auto target_drivable_lanes =
+    getNonOverlappingExpandedLanes(*reference_path, drivable_lanes);
+
+  // for old architecture
+  utils::generateDrivableArea(
+    *reference_path, target_drivable_lanes, p.vehicle_length, planner_data_);
+
+  // 3. create obstacle polygons to avoid
+  std::vector<tier4_autoware_utils::Polygon2d> obstacle_polys;
+  for (const auto & object : target_objects_) {
+    const auto obstacle_poly = calcDynamicObstaclePolygon(*reference_path, object);
+    if (obstacle_poly) {
+      obstacle_polys.push_back(obstacle_poly.value());
+    }
+  }
+
+  BehaviorModuleOutput output;
+  output.path = reference_path;
+  output.reference_path = reference_path;
```

Comment:
I think it should use previous module's turn signal info.

Reply:
DONE
https://github.com/autowarefoundation/autoware.universe/pull/3415/commits/c3ffd01951319c61b10c7f4534cfd620b0227851